cassandra/src/main/scala/filodb.cassandra/columnstore/CassandraColumnStore.scala

```diff
@@ -291,68 +292,33 @@ trait CassandraChunkSource extends RawChunkSource with StrictLogging {
 
   def readRawPartitions(ref: DatasetRef,
+                        maxChunkTime: Long,
                         partMethod: PartitionScanMethod,
                         chunkMethod: ChunkScanMethod = AllChunkScan): Observable[RawPartData] = {
-    val indexTable = getOrCreateIngestionTimeIndexTable(ref)
-    logger.debug(s"Scanning partitions for $ref with method $partMethod...")
-    val (filters, infoRecords) = partMethod match {
-      case SinglePartitionScan(partition, _) =>
-        (Nil, indexTable.getInfos(partition))
-
-      case MultiPartitionScan(partitions, _) =>
-        (Nil, indexTable.getMultiInfos(partitions))
-
-      case FilteredPartitionScan(CassandraTokenRangeSplit(tokens, _), filters) =>
-        (filters, indexTable.scanInfos(tokens))
-
-      case other: PartitionScanMethod =>  ???
-    }
-    // For now, partition filter func is disabled.  In the near future, this logic to scan through partitions
-    // and filter won't be necessary as we will have Lucene indexes to pull up specific partitions.
-    // val filterFunc = KeyFilter.makePartitionFilterFunc(dataset, filters)
-
+    val chunkTable = getOrCreateChunkTable(ref)
     partMethod match {
-      // Compute minimum start time from the infos, then do efficient range query MULTIGET
-      case MultiPartitionScan(partitions, _) =>
-        val chunkTable = getOrCreateChunkTable(ref)
-        Observable.fromTask(startTimeFromInfos(infoRecords, chunkMethod.startTime, chunkMethod.endTime))
-          .flatMap { case startTime =>
-            chunkTable.readRawPartitionRange(partitions, startTime, chunkMethod.endTime)
-          }
-      case other: PartitionScanMethod =>
-        // NOTE: we use hashCode as an approx means to identify ByteBuffers/partition keys which are identical
-        infoRecords.sortedGroupBy(_.binPartition.hashCode)
-                   .mapAsync(partParallelism) { case (_, binIndices) =>
-                     val infoBytes = binIndices.map(_.data.array)
-                     assembleRawPartData(ref, binIndices.head.binPartition.array, infoBytes, chunkMethod)
-                   }
-    }
-  }
-
-  // Returns the minimum start time from all the ChunkInfos that are within [startTime, endTime) query range
-  // Needed so we can do range queries efficiently by chunkID
-  private def startTimeFromInfos(infoRecords: Observable[InfoRecord],
-                                 startTime: Long,
-                                 endTime: Long): Task[Long] = {
-    infoRecords.foldLeftL(Long.MaxValue) { case (curStart, InfoRecord(_, data)) =>
-      val infoStartTime = ChunkSetInfo.getStartTime(data.array)
-      val infoEndTime = ChunkSetInfo.getEndTime(data.array)
-      if (infoStartTime <= endTime && infoEndTime >= startTime) {
-        Math.min(curStart, infoStartTime)
-      } else {
-        curStart
-      }
+      case FilteredPartitionScan(CassandraTokenRangeSplit(tokens, _), Nil)  =>
+        chunkTable.scanPartitionsBySplit(tokens)
```

Comment:
Shouldn't we also pass in the start and end time?  C* is able to filter by partition key range as well, which would limit the scan time of this query significantly.

Reply:
We use FilteredPartitionScan sparingly and only in tests without supporting filters - this should be a different PR with different priority.